spec.html

```diff
@@ -42955,6 +42966,383 @@ <h1>Properties of _TypedArray_ Instances</h1>
       <p>_TypedArray_ instances are TypedArrays. Each _TypedArray_ instance inherits properties from the corresponding _TypedArray_ prototype object. Each _TypedArray_ instance has the following internal slots: [[ViewedArrayBuffer]], [[TypedArrayName]], [[ContentType]], [[ByteLength]], [[ByteOffset]], and [[ArrayLength]].</p>
     </emu-clause>
   </emu-clause>
+
+  <emu-clause id="sec-uint8array">
+    <h1>Uint8Array Objects</h1>
+    <p>A Uint8Array is a particular kind of _TypedArray_ as described above. In addition, there are certain additional methods on the Uint8Array constructor and on the Uint8Array prototype object.</p>
+
+    <emu-clause id="sec-additional-properties-of-the-uint8array-constructor">
+      <h1>Additional Properties of the Uint8Array Constructor</h1>
+
+      <emu-clause id="sec-uint8array.frombase64">
+        <h1>Uint8Array.fromBase64 ( _string_ [ , _options_ ] )</h1>
+        <emu-alg>
+          1. If _string_ is not a String, throw a *TypeError* exception.
+          1. Let _opts_ be ? GetOptionsObject(_options_).
+          1. Let _alphabet_ be ? Get(_opts_, *"alphabet"*).
+          1. If _alphabet_ is *undefined*, set _alphabet_ to *"base64"*.
+          1. If _alphabet_ is neither *"base64"* nor *"base64url"*, throw a *TypeError* exception.
+          1. Let _lastChunkHandling_ be ? Get(_opts_, *"lastChunkHandling"*).
+          1. If _lastChunkHandling_ is *undefined*, set _lastChunkHandling_ to *"loose"*.
+          1. If _lastChunkHandling_ is not one of *"loose"*, *"strict"*, or *"stop-before-partial"*, throw a *TypeError* exception.
+          1. Let _result_ be FromBase64(_string_, _alphabet_, _lastChunkHandling_).
+          1. If _result_.[[Error]] is not ~none~, then
+            1. Throw _result_.[[Error]].
+          1. Let _resultLength_ be the length of _result_.[[Bytes]].
+          1. Let _ta_ be ? <emu-meta suppress-effects="user-code">AllocateTypedArray(*"Uint8Array"*, %Uint8Array%, *"%Uint8Array.prototype%"*, _resultLength_)</emu-meta>.
+          1. Set the value at each index of _ta_.[[ViewedArrayBuffer]].[[ArrayBufferData]] to the value at the corresponding index of _result_.[[Bytes]].
```

Comment:
Is there any way that user code in the previous step could cause `_ta_.[[ViewedArrayBuffer]].[[ArrayBufferData]]` or `_result_.[[Bytes]]` to change, possibly in a way that causes them to be different sizes? If not, let's add an assertion before this step that they are definitely the same size.